engine/consensus/dkg/processor/engine_test.go

```diff
@@ -202,3 +202,66 @@ func TestBroadcastMessage(t *testing.T) {
 	engine.Broadcast(msgb)
 	contractClient.AssertExpectations(t)
 }
+
+// TestReadMessages checks that the engine correctly calls the smart contract
+// to fetch broadcast messages, and forwards the messages to the msgCh.
+func TestReadBroadcastMessages(t *testing.T) {
+
+	engine := createTestEngine(t, committee[orig])
+	blockID := unittest.IdentifierFixture()
+	expectedMsgs := []msg.DKGMessage{
+		msg.NewDKGMessage(
+			orig,
+			[]byte("message 1"),
+			engine.dkgInstanceID,
+		),
+		msg.NewDKGMessage(
+			orig,
+			[]byte("message 2"),
+			engine.dkgInstanceID,
+		),
+		msg.NewDKGMessage(
+			orig,
+			[]byte("message 3"),
+			engine.dkgInstanceID,
+		),
+	}
+
+	// check that the dkg contract client is called correctly
+	contractClient := &module.DKGContractClient{}
+	contractClient.On("ReadBroadcast", engine.messageOffset, blockID).
+		Return(expectedMsgs, nil).
+		Once()
+	engine.dkgContractClient = contractClient
+
+	// launch a background routine to capture messages forwarded to the msgCh
+	receivedMsgs := []msg.DKGMessage{}
+	doneCh := make(chan struct{})
+	go func() {
+		for {
+			msg := <-engine.msgCh
+			receivedMsgs = append(receivedMsgs, msg)
+			if len(receivedMsgs) == len(expectedMsgs) {
+				close(doneCh)
+			}
+		}
+	}()
+
+	err := engine.fetchBroadcastMessages(blockID)
+	require.NoError(t, err)
+
+	// check that the contract has been correctly called
+	contractClient.AssertExpectations(t)
+
+	// check that the messages have been received and forwarded to the msgCh
+	unittest.AssertReturnsBefore(
+		t,
+		func() {
+			<-doneCh
+		},
+		time.Second)
```

Comment:
```suggestion
	unittest.AssertClosesBefore(t, doneCh, time.Second)
```

https://github.com/onflow/flow-go/blob/2307e5a2006e05c7b0f0f656bbbbcc12b60b13ea/utils/unittest/unittest.go#L48